server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -547,16 +549,11 @@ private String getPrettyName() throws IOException {
             final Optional<String> maybePrettyNameLine =
                     prettyNameLines.size() == 1 ? Optional.of(prettyNameLines.get(0)) : Optional.empty();
             if (maybePrettyNameLine.isPresent()) {
-                final String prettyNameLine = maybePrettyNameLine.get();
-                final String[] prettyNameFields = prettyNameLine.split("=");
-                assert prettyNameFields.length == 2 : prettyNameLine;
-                if (prettyNameFields[1].length() >= 3 &&
-                        (prettyNameFields[1].startsWith("\"") && prettyNameFields[1].endsWith("\"")) ||
-                        (prettyNameFields[1].startsWith("'") && prettyNameFields[1].endsWith("'"))) {
-                    return prettyNameFields[1].substring(1, prettyNameFields[1].length() - 1);
-                } else {
-                    return prettyNameFields[1];
-                }
+                // we trim since some OS contain trailing space, for example, Oracle Linux Server 6.9 has a trailing space after the quote
+                final String trimmedPrettyNameLine = maybePrettyNameLine.get().trim();
+                final Matcher matcher = Pattern.compile("PRETTY_NAME=(\"?|'?)?([^\"']+)\\1\\s*").matcher(trimmedPrettyNameLine);
```

Comment:
If we're calling `trim()` do we need the trailing `\\s*` in the regex as well?

Reply:
Oops, that's leftover from my first approach. I pushed 15ce2a0cbad8f97526d63d21341449015b8f581a.